beacon-chain/rpc/lookup/stater.go

```diff
@@ -143,8 +143,7 @@ func (p *BeaconDbStater) State(ctx context.Context, stateId []byte) (state.Beaco
 			return nil, errors.Wrap(err, "could not get justified state")
 		}
 	default:
-		stringId := strings.ToLower(string(stateId))
-		if len(stringId) >= 2 && stringId[:2] == "0x" {
```

Comment:
This is a logic change, not just a redundant conversion. Previously `0X` and `0x` prefixes would both match, now only `0x` matches.

Reply:
Ah, didn't see that. Thanks